content/blog/opa-eng-plugin/opa-eng-plugin.md

```diff
@@ -0,0 +1,63 @@
+---
+title: You can use OPA for Authorization in Backstage!
+date: "2024-02-03T12:57:59Z"
+description: "Part 1: How to use OPA with Backstage for authorization"
+---
+
+Hello everyone! In this post we are going to see how you can use OPA with Backstage for authorization!
+
+## What is OPA?
+
+[OPA](https://www.openpolicyagent.org/) (pronounced “oh-pa”) is an open source, general-purpose policy engine that unifies policy enforcement across the stack. OPA is an open source project that was created by [Styra](https://www.styra.com/) and then donated to the [Cloud Native Computing Foundation](https://www.cncf.io/). It uses the declarative language Rego to define policies, more information on that [here](https://www.openpolicyagent.org/docs/latest/policy-language/).
+
+## What is Backstage?
+
+[Backstage](https://backstage.io/) is an open source framework for creating a developer portal (IDP or Internal Developer Portal). Backstage was created by Spotify and donated to the [Cloud Native Computing Foundation](https://www.cncf.io/) and has gained a lot of traction in the open source community.
+
+## Why use OPA with Backstage?
+
+Backstage uses the [Permissions Framework](https://backstage.io/docs/permissions/overview) to define permissions, this is very useful for defining permissions like RBAC in the portal, the permissions are defined in 'policies' that are written in TypeScript. This in itself is not bad and for most will be fine, but it does have some drawbacks, for example:
+
+- If you don't know TypeScript, you can't define permissions.
+- If you have a security team that is responsible for defining permissions, they have to learn TypeScript to be able to define permissions.
+- The policies can be difficult to maintain, especially if they are large.
+- You need to redeploy the code every time you change the permissions.
```

Comment:
I feel like my list might be more like:

- Permissions must be encoded in Backstage specially, rather than using a generic system.
- Updating policy code requires redeployment.
- Larger policies are harder to maintain and test.

You might also want to mention auditing here, but that is perhaps also something that BS supports and so less of an advantage.